ios/PolyPodApp/Core/PolyPodCore.swift

```diff
@@ -0,0 +1,51 @@
+// Please remove this line and the empty one after it
+
+import Foundation
+import PolyPodCore
+
+final class PolyPodCore {
+    static let instance = PolyPodCore()
+    let corePointer: OpaquePointer
+    
+    private let containerPath: UnsafePointer<CChar>
+    
+    init() {
+        // TODO: Handle possible errors
+        let path = try! FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: false).path
+        containerPath = NSString(string: path).utf8String!
+        
+        corePointer = new_poly_pod_core(.default(containerPath))
+    }
+}
+
+extension PolyPodCore {
+    var showInApp: Bool {
+        update_notification_show_in_app(corePointer)
+    }
+    
+    var showPush: Bool {
+        update_notification_show_push(corePointer)
+    }
+
+    func handleStartup() {
+        update_notification_handle_startup(corePointer)
+    }
+    
+    func handlePushSeen() {
+        update_notification_handle_push_seen(corePointer)
+    }
+    
+    func handleFirstRun() {
+        update_notification_handle_first_run(corePointer)
+    }
+    
+    func handleInAppSeen() {
+        update_notification_handle_in_app_seen(corePointer)
+    }
```

Comment:
Seeing all this boilerplate, I think this approach might be OK to start with, but I'm wondering if we couldn't have a more generic interface... Something like:

```
bool show_push = polypod_core_call_bool_method(core_pointer, "update_notification.show_push", NO_ARGS);
polypod_core_call_void_method(core_pointer, "update_notification.handle_push_seen", NO_ARGS);
```

We might need to write some C code for reasonably convenient usage of that, but at least we wouldn't have to touch all platforms for every single signature change...

Reply:
By boilerplate I guess you are referring to code like this:
```
func handleFirstRun() {
     update_notification_handle_first_run(corePointer)
}
```
It is not mandatory to have it like this. I mostly added this with the idea to hide the pointer related stuff.
But, this:
```
PolyPodCore.instance.handleFirstRun()
```
is identical to this:
```
update_notification_handle_first_run(PolyPodCore.instance.corePointer)
```
Still I wouldn't spread this throughout PolyPod, it will be harder for testability.

Ideally we should have a protocol in swift, like UpdateNotification describing the functionality PolyPod expects, and PolyPodCore would implement it, ending with the code we have right now:

```
protocol UpdateNotification {
    var showInApp: Bool
    var showPush: Bool
    func handleStartup()
    ....
}

extension PolyPodCore: UpdateNotifiction { ... }
```

I was playing with this idea, to have a String based API so to say. When PolyPod would send messages to Core with some payload. I guess somewhat similar to PodAPI. But I don't like idea of "encoding" something just to "decode" it other end straight away, when we can have a strongly typed API.

Reply:
Yeah, good points. I think we can:

1. Follow through like you did it in this PR until it becomes painful (at which point the better alternative should be more obvious)
2. Generally stick to the plan of trying to make the communication "surface" between platform and core as thin as possible